include/functions/math.hpp

```diff
@@ -125,6 +124,60 @@ BigInt sqrt(const BigInt& num) {
     return sqrt_current;
 }
 
+/*
+   is_probable_prime
+   -----------------
+   Returns bool that indicates the primality of a BigInt based 
+   on primality testing using Rabin-Miller testing
+   NOTE: certainty determines how accurate test is and must be greater than 0
+   			
+ */
+
+bool BigInt::is_probable_prime(size_t certainty) const {
+    BigInt num = this->value;
+
+    // Covering corner cases
+    if (num <= 1 or num == 4) return false;
+    if (num <= 3) return true;
+    
+    // Find r such that n = 2^d * r + 1 for some r >= 1
```

Comment:
Is `n` = `num` and `r` = `res`? If so, then please make sure that the names in the formula and in the code are the same in order to explain the code better.